runtime/lua/vim/lsp/inlay_hint.lua

```diff
@@ -0,0 +1,222 @@
+      if type(hint.label) == 'table' and hint.label.command then
+        countdown()
+      else
+        client.request('inlayHint/resolve', hint, function(_, result)
+          if result and type(result.label) == 'table' and result.label.command then
+            hint.label.command = result.label.command
+          end
+          num_unresolved = num_unresolved - 1
+          if num_unresolved == 0 and current_line_hints then
+            hints_by_lnum[lnum] = new_line_hints
+          end
+          countdown()
+        end)
+      end
+    end
+  end
+  if redraw then
+    vim.cmd('redraw!')
+  end
+end
+
+--- |lsp-handler| for the method `textDocument/inlayHint`
+---
+function M.on_inlayhint(err, result, ctx, _)
+  if err then
+    local _ = log.error() and log.error('inlayhint', err)
+    return
+  end
+  M.save(result, ctx.bufnr, ctx.client_id, ctx.version)
+end
+
+--- Refresh the inlay hints for the current buffer
+---
+--- It is recommended to trigger this using an autocmd or via keymap.
+---
+--- Example:
+--- <pre>vim
+---   autocmd BufEnter,InsertLeave,BufWritePost <buffer> lua vim.lsp.inlay_hint.refresh()
+--- </pre>
+---
+function M.refresh()
+  local params = {
+    textDocument = util.make_text_document_params(),
+    range = {
+      start = { line = 0, character = 0 },
+      ['end'] = { line = vim.api.nvim_buf_line_count(0), character = 0 },
+    },
+  }
+  vim.lsp.buf_request(0, 'textDocument/inlayHint', params)
+end
+
+---@private
+local function resolve_bufnr(bufnr)
+  return bufnr == 0 and api.nvim_get_current_buf() or bufnr
+end
+
+--- Clear the hints
+---
+---@param client_id integer|nil filter by client_id. All clients if nil
+---@param bufnr integer|nil filter by buffer. All buffers if nil
+function M.clear(client_id, bufnr)
+  local buffers = bufnr and { resolve_bufnr(bufnr) } or vim.tbl_keys(hint_cache_by_buf)
+  for _, iter_bufnr in pairs(buffers) do
+    local bufstate = hint_cache_by_buf[iter_bufnr]
+    local client_lens = (bufstate or {}).client_hint or {}
+    local client_ids = client_id and { client_id } or vim.tbl_keys(client_lens)
+    for _, iter_client_id in pairs(client_ids) do
+      if bufstate then
+        bufstate.client_hint[iter_client_id] = {}
+      end
+    end
+  end
+  vim.cmd('redraw!')
+end
+
+local namespace = api.nvim_create_namespace('vim_lsp_inlayhint')
+
+api.nvim_set_decoration_provider(namespace, {
+  on_win = function(_, _, bufnr, topline, botline)
+    local bufstate = hint_cache_by_buf[bufnr]
+    if not bufstate then
+      return
+    end
+
+    if bufstate.version ~= vim.lsp.util.buf_versions[bufnr] then
+      return
+    end
+    local hints_by_client = bufstate.client_hint
+    api.nvim_buf_clear_namespace(bufnr, namespace, 0, -1)
+
+    for i = topline, botline do
+      for _, hints_by_lnum in pairs(hints_by_client) do
+        local line_hints = hints_by_lnum[i] or {}
+        for _, hint in pairs(line_hints) do
+          local text
```

Comment:
```suggestion
          local label = type(hint.label) == 'string' and {hint.label}
          local text = ''
          for _ , part in ipairs(label) do
             text = part.value and part or text ..part.value
          end            
```

Reply:
If I understand correctly, that won't work as the type of `hint.label` is either `string` or `table`, and you iterate through `hint.label` only if it is `table`

Reply:
you check is `hint.label` is string or table. if string use `hint.label` if table. use value . this code wrap label to table then do once iterator if it's string this for only run once. did you test it ?

Reply:
yes, `vim_lsp_inlayhint: Error executing lua: ...f/pkgbuilds/nvim/repo/runtime/lua/vim/lsp/inlay_hint.lua:194: attempt to concatenate field 'value' (a nil value)`

lua-ls doesn't like it either

Reply:
sry i had changed it before has a wrong place

Reply:
tip: for a `list-like` table use `iparis` not `pairs` .

Reply:
This works but is harder to read, can you please explain why I should do that?

Reply:
personal code quirk. Reduce ifelse machine instructions. abstract the behavior into a public behavior . Too many if for nested. It will only make the code execution time increase linearly with the length of the table O(n)， readability. It's fine for me. maybe because I wrote it. :) if you don't want change you can ignore